.github/workflows/dco.yml

```
@@ -0,0 +1,45 @@
+name: Test Developer Certificate of Origin
+
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: Test Developer Certificate of Origin
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+          persist-credentials: false
+      - name: Determine branch name
+        run: |
+          BRANCH="${GITHUB_BASE_REF#refs/heads/}"
+          echo "Checking DCO for every commit on branch ${BRANCH}"
+          echo "BRANCH=${BRANCH}" >> ${GITHUB_ENV}
+      - name: Test DCO for every commit
+        run: |
+          RET=0
+          SEEN_COMMITS=0
+          # this will not work properly if what we are testing is not a pull request
+          for commit in $(git rev-list HEAD ^origin/${BRANCH}); do
+            echo "=== Checking commit '${commit}'"
+            body="$(git show -s --format=%b ${commit})"
+            expected="$(git show -s --format='Signed-off-by: %cN <%cE>' ${commit})"
```

You can use trailers to only get the `Signed-off-by` value(s):
```suggestion
            body="$(git show -s --format='%(trailers:key=Signed-off-by,valueonly)\ ${commit})"
            expected="$(git show -s --format='%cN <%cE>' ${commit})"
```

https://stackoverflow.com/questions/31261884/git-detect-if-commit-is-signed-off mentions this needs Git 2.31, which is from 2021 so by now you can probably count on it to be supported widely.